client/src/components/NavBar/NavBar.js

```diff
@@ -27,6 +30,18 @@ function NavBar(props) {
     setShow(true);
     sessionStorage.clear();
   }; //clear session storage to wipe the current Cart on user change
+  const [overdueItems, setOverdueItems] = useState(0); // State for storing the number of overdue items
+
+  const getInfo = async () => {
+    let allRecords = await getAllRecords();
```

Comment:
Instead of getting all records from the database, just get the checked-out records, then cound the overdue from that. Recrods is the largest table. THere is already a controller for it.
```suggestion
    let allRecords = await getAllCheckedOutRecords();
```